src/qt/configurenamedialog.cpp

```diff
@@ -61,7 +64,15 @@ void ConfigureNameDialog::accept()
     returnData = ui->dataEdit->text();
     returnTransferTo = ui->transferTo->text();
 
-    QDialog::accept();
+    if(!IsValidJSON(data) | data != ""){
```

Comment:
That `|` should be `||` (logical OR instead of bitwise OR).

I personally would also prefer `!data.empty ()` over `data != ""`, but that is a matter of style and both are fine with me.

Generally speaking though, what exactly is the intent of the condition?  I would find it logical if it was `!IsValidJSON(data) && !data.empty()` instead of OR, i.e. the "accepted" data is valid JSON or the empty string (since that might be a common use of invalid JSON).

Reply:
Since this logic shows up twice in this file, would it be OK to move the empty-string special-case logic into the `IsValidJSON` function? I realize this is maybe semantically misleading since an empty string is not actually valid JSON, but since that function lives in `applications.cpp` and I can't think of literally any contexts where application-layer values that need a JSON object would need to know the difference between an empty JSON object an an empty string, I think this would reduce the risk of accidents if we forget to copy over the special-case logic into future code that calls that function.

Reply:
Yes that makes sense to me.  Maybe call the function then `IsValidJSONOrEmpty` just to avoid any confusion.